cmd/ingester/app/consumer/consumer.go

```diff
@@ -46,16 +47,14 @@ type Consumer struct {
 
 	deadlockDetector deadlockDetector
 
-	partitionIDToState  map[int32]*consumerState
 	partitionMapLock    sync.Mutex
 	partitionsHeld      int64
```

Comment:
if the map is removed, the mutex is not particularly useful
```suggestion
	partitionsHeld      atomic.Int64
```

Reply:
OK， see commit：https://github.com/jaegertracing/jaeger/pull/4294/commits/ec4e958c613badbbecc825eba30fca306ac39396